PWGHF/TableProducer/reducedDataCreatorDplusPi.cxx

```diff
@@ -0,0 +1,483 @@
+
+    // Initialize fitter
+    df3.setPropagateToPCA(propagateToPCA);
+    df3.setMaxR(maxR);
+    df3.setMaxDZIni(maxDZIni);
+    df3.setMinParamChange(minParamChange);
+    df3.setMinRelChi2Change(minRelChi2Change);
+    df3.setUseAbsDCA(useAbsDCA);
+    df3.setWeightedFinalPCA(useWeightedFinalPCA);
+
+    // Configure CCDB access
+    ccdb->setURL(ccdbUrl);
+    ccdb->setCaching(true);
+    ccdb->setLocalObjectValidityChecking();
+    lut = o2::base::MatLayerCylSet::rectifyPtrFromFile(ccdb->get<o2::base::MatLayerCylSet>(ccdbPathLut));
+    if (!o2::base::GeometryManager::isGeometryLoaded()) {
+      ccdb->get<TGeoManager>(ccdbPathGeo);
+    }
+    runNumber = 0;
+  }
+
+  /// Pion selection (D Pi <-- B0)
+  /// \param trackPion is a track with the pion hypothesis
+  /// \param track0 is prong0 of selected D candidate
+  /// \param track1 is prong1 of selected D candidate
+  /// \param track2 is prong2 of selected D candidate
+  /// \param indexD is an array of globalIndex of each D prong
+  /// \return true if trackPion passes all cuts
+  template <typename T1, typename T2>
+  bool isPionSelected(const T1& trackPion, const T2& track0, const T2& track1, const T2& track2)
+  {
+    // check isGlobalTrackWoDCA status for pions if wanted
+    if (usePionIsGlobalTrackWoDCA && !trackPion.isGlobalTrackWoDCA()) {
+      return false;
+    }
+    // minimum pT selection
+    if (trackPion.pt() < ptPionMin || !isSelectedTrackDCA(trackPion)) {
+      return false;
+    }
+    // reject pions that are D daughters
+    if (trackPion.globalIndex() == track0.globalIndex() || trackPion.globalIndex() == track1.globalIndex() || trackPion.globalIndex() == track2.globalIndex()) {
+      return false;
+    }
+    // reject pi D with same sign as D
+    if (trackPion.sign() * track0.sign() > 0) {
+      return false;
+    }
+    return true;
+  }
+
+  /// Single-track cuts for pions on dcaXY
+  /// \param track is a track
+  /// \return true if track passes all cuts
+  template <typename T>
+  bool isSelectedTrackDCA(const T& track)
+  {
+    auto pTBinTrack = findBin(binsPtPion, track.pt());
+    if (pTBinTrack == -1) {
+      return false;
+    }
+
+    if (std::abs(track.dcaXY()) < cutsTrackPionDCA->get(pTBinTrack, "min_dcaxytoprimary")) {
+      return false; // minimum DCAxy
+    }
+    if (std::abs(track.dcaXY()) > cutsTrackPionDCA->get(pTBinTrack, "max_dcaxytoprimary")) {
+      return false; // maximum DCAxy
+    }
+    return true;
+  }
+
+  void process(aod::Collisions const& collisions,
+               CandsDFiltered const& candsD,
+               aod::TrackAssoc const& trackIndices,
+               TracksPIDWithSel const&,
+               aod::BCsWithTimestamps const&)
+  {
+    // store configurables needed for B0 workflow
+    if (!isHfCandB0ConfigFilled) {
+      int mySelectionFlagD = selectionFlagD;
+      rowCandidateConfig(mySelectionFlagD);
```

Comment:
Why is this not filled directly?

Reply:
I remember trying this in the past and I got errors because I was filling a table with a configurable. That's why I define a new variable of type `int` and not `Configurable<int>` to fill the table.

Reply:
Have you tried `selectionFlagD.value`?

Reply:
No, I had not tried it and it works.